docs/tutorials/dynamics_backend.rst

```diff
@@ -0,0 +1,489 @@
+jax>` for more information.
+
+.. jupyter-execute::
+    :hide-code:
+
+    import warnings
+    warnings.filterwarnings('ignore', message='', category=Warning, module='', lineno=0, append=False)
+
+.. jupyter-execute::
+
+    # Configure to use JAX internally
+    import jax
+    jax.config.update("jax_enable_x64", True)
+    jax.config.update("jax_platform_name", "cpu")
+    from qiskit_dynamics.array import Array
+    Array.set_default_backend("jax")
+
+2. Instantiating a minimally-configured :class:`.DynamicsBackend` with a 2 qubit model
+--------------------------------------------------------------------------------------
+
+To create the :class:`.DynamicsBackend`, first specify a :class:`.Solver` instance using the model
+details. For the model we will use a 2 qubit transmon model, with Hamiltonian:
+
+.. math:: 
+    
+    H(t) = 2 \pi \nu_0 &N_0 + \pi \alpha_0 N_0 (N_0 - I) + 2 \pi \nu_1 N_1
+    + \pi \alpha_1 N_1(N_1 - I) + 2 \pi J (a_0 + a_0^\dagger)(a_1 + a_1^\dagger) \\ 
+    & + 2 \pi r_0 s_0(t)(a_0 + a_0^\dagger) + 2 \pi r_1 s_1(t)(a_1 + a_1^\dagger),
+
+where 
+
+- :math:`\nu_0` and :math:`\nu_1` are the qubit frequencies, 
+- :math:`\alpha_0` and :math:`\alpha_1` are the qubit anharmonicities, 
+- :math:`J` is the coupling strength, 
+- :math:`r_0` and :math:`r_1` are the Rabi strengths, and :math:`s_0(t)` and :math:`s_1(t)` are the
+  drive signals, 
+- :math:`a_j` and :math:`a_j^\dagger` are the lowering and raising operators for qubit :math:`j`,
+  and
+- :math:`N_0` and :math:`N_1` are the number operators.
+
+.. jupyter-execute::
+
+    import numpy as np
+    
+    dim = 3
+    
+    v0 = 4.86e9
+    anharm0 = -0.32e9
+    r0 = 0.22e9
+    
+    v1 = 4.97e9
+    anharm1 = -0.32e9
+    r1 = 0.26e9
+    
+    J = 0.002e9
+    
+    a = np.diag(np.sqrt(np.arange(1, dim)), 1)
+    adag = np.diag(np.sqrt(np.arange(1, dim)), -1)
+    N = np.diag(np.arange(dim))
+    
+    ident = np.eye(dim, dtype=complex)
+    full_ident = np.eye(dim**2, dtype=complex)
+    
+    N0 = np.kron(ident, N)
+    N1 = np.kron(N, ident)
+    
+    a0 = np.kron(ident, a)
+    a1 = np.kron(a, ident)
+    
+    a0dag = np.kron(ident, adag)
+    a1dag = np.kron(adag, ident)
+    
+    
+    static_ham0 = 2 * np.pi * v0 * N0 + np.pi * anharm0 * N0 * (N0 - full_ident)
+    static_ham1 = 2 * np.pi * v1 * N1 + np.pi * anharm1 * N1 * (N1 - full_ident)
+    
+    static_ham_full = static_ham0 + static_ham1 + 2 * np.pi * J * ((a0 + a0dag) @ (a1 + a1dag))
+    
+    drive_op0 = 2 * np.pi * r0 * (a0 + a0dag)
+    drive_op1 = 2 * np.pi * r1 * (a1 + a1dag)
+
+Construct the :class:`.Solver` using the model details, including parameters necessary for pulse
+simulation. See the :class:`.Solver` documentation, as well as the :ref:`tutorial example <pulse
+solver>` for more details.
+
+.. jupyter-execute::
+
+    from qiskit_dynamics import Solver
+    
+    # build solver
+    dt = 1/4.5e9
+    
+    solver = Solver(
```

Comment:
To me it looks bit weird (and confusing) it is called _solver_. These constructor arguments make me think it is a _model_ rather than a solver, and solver is something that consumes the model with solver options (run options in terms of qiskit) -- but I know the solver functionality is indeed integrated into this class. This is an issue of class design and not the scope of this PR -- but maybe you can update the dynamics backend (e.g. `DynamicBackend.from_hamiltonian`) so that it can take these parameters and internally instantiate the solver instance to hide implementation details.

Reply:
I agree that the `Solver` terminology is a bit weird. The core design of dynamics (from the first release) is model classes, `HamiltonianModel`, `LindbladModel`, ..., which you then pass to the core solver functions `solve_ode` and `solve_lmde`. This ended up being slightly awkward to use however, so we made the `Solver` class as a high level wrapper of these things (it builds the right kind of model internally, then calls the underlying solver functions with it). I think in the future, as we refactor/refine the package, this awkwardness should be addressed. I have some ideas for how this might look, but yeah as you say it is well outside the scope of this PR :).

For the `DynamicsBackend.from_hamiltonian` comment: we did discuss this earlier in the `DynamicsBackend` development process. The conclusion was that, if we made the constructor take in all of the model arguments directly, it would be very redundant. I.e. it ends up looking exactly like the constructor for `Solver` (which can take a lot of arguments), plus only a few other arguments. I think this had the effect of obscuring what was different between the `Solver` and `DynamicsBackend` constructors, so we ended up just deciding to have the main constructor for `DynamicsBackend` consume a `Solver` for clarity, and to avoid replicating the `Solver.__init__` doc string. I realize you're suggesting something slightly different: to add another static method that does this, but I think the reasoning still applies.

Reply:
There are more computations being done by `DynamicsBackend` (the measurement portion), but yeah `Solver` is the main thing.